### PR TITLE
dc6 editor: merge frames feature

### PR DESCRIPTION
**Pull request by gucio321**

### CODE REVIEW

**Comment by gravestench** [comment]
what does this feature do?

**Comment by gucio321** [comment]
@gravestench
For example, if you have a game sprite (usually some game background) in DC6 animation format  it is written in a few frames.
this feature allows user to merge this frames and view all image.

**Review by gravestench — CHANGES_REQUESTED** [review]
Can we change the name of this feature to "TiledView" or or "Tiled" or something?

File: hswidget/dc6widget/state.go [hunk 1]
```diff
@@ -46,11 +47,13 @@ type widgetMode int32
 
 const (
 	dc6WidgetViewer widgetMode = iota
+	dc6WidgetTiledView
 )
 
 // widgetState represents dc6 viewer's state
 type widgetState struct {
 	viewerState
+	mergeState
```
Comment (gravestench) [inline hswidget/dc6widget/state.go]: change to `tiledState`

File: hswidget/dc6widget/widget.go [hunk 2]
```diff
@@ -156,6 +164,24 @@ func (p *widget) makePlayerLayout(state *widgetState) giu.Layout {
...
+			giu.Label("Tiled view:"),
+			giu.InputInt("Width##"+p.id+"mergeWidth", &state.width).Size(inputIntW).OnChange(func() {
+				p.recalculateTiledViewHeight(state)
+			}),
+			giu.InputInt("Height##"+p.id+"mergeHeight", &state.height).Size(inputIntW).OnChange(func() {
+				p.recalculateTiledViewWidth(state)
+			}),
+		),
+		giu.Image(state.merged).Size(float32(state.imgw), float32(state.imgh)),
+		giu.Button("Back##"+p.id+"mergeBack").Size(buttonW, buttonH).OnClick(func() {
```
Comment (gravestench) [inline hswidget/dc6widget/widget.go]: change `merge` to `tiled`

File: hswidget/dc6widget/widget.go [hunk 3]
```diff
@@ -117,13 +124,14 @@ func (p *widget) makeViewerLayout() giu.Layout {
...
 		giu.Separator(),
-		// NOTE: most of DC6 animations arent `playable`
-		// they are sprites, however exists some animations
-		// for example monsters' animations, where `play`
-		// feature is useful
 		p.makePlayerLayout(viewerState),
 		giu.Separator(),
 		widget,
+		giu.Separator(),
+		giu.Button("Tiled View##"+p.id+"mergebutton").Size(buttonW, buttonH).OnClick(func() {
```
Comment (gravestench) [inline hswidget/dc6widget/widget.go]: for coherence, change `mergebutton` to `tiledButton`

File: hswidget/dc6widget/state.go [hunk 4]
```diff
@@ -177,6 +197,19 @@ func (s *viewerState) Dispose() {
...
+	merged *giu.Texture
+	imgw,  // nolint:structcheck // linter's bug - it is used
+	imgh int
+}
+
+func (s *mergeState) Dispose() {
+	s.width, s.height = 0, 0
+	s.merged = nil
+}
+
```
Comment (gravestench) [inline hswidget/dc6widget/state.go]: change `merge` to `tiled`

**Review by gravestench — CHANGES_REQUESTED** [review]
change all occurrences of `merge` to `tiled`

File: hswidget/dc6widget/state.go (hunk 4 above)
Comment (gucio321) [reply hswidget/dc6widget/state.go]: ecefa8

File: hswidget/dc6widget/widget.go (hunk 3 above)
Comment (gucio321) [reply hswidget/dc6widget/widget.go]: fixed

**Comment by gucio321** [comment]
> change all occurrences of `merge` to `tiled`

I'm sorry, I searched only for `/Merged` and forgot about `/merge` :grinning:
Thanks for reviews!

**Review by gravestench — APPROVED** [review]

**Pull request merged**